### ISSUE TITLE
Grape dependent

### ISSUE THREAD

**Issue by vypah** [ruby-grape/grape#1599]

**Comment by dblock** [comment]
Did you try this?

**Comment by vypah** [comment]
@dblock yes. it just bails out with Grape Exception of `Unknown Parameter`

**Comment by dblock** [comment]
I labeled this as feature request, contributions welcome!

**Comment by vypah** [comment]
@dblock After having an initial look at the source, I agree that this is more like a feature request as the given method is not suppose to handle this.

```
def given(*attrs, &block)
        attrs.each do |attr|
          proxy_attr = attr.is_a?(Hash) ? attr.keys[0] : attr
          raise Grape::Exceptions::UnknownParameter.new(proxy_attr) unless declared_param?(proxy_attr)
        end
        new_lateral_scope(dependent_on: attrs, &block)
end
```

Especially this `dependent_on` param passed to `new_lateral_scope` totally kills this scenario as there is no dependent param in this case. What I would suggest is to have a separate implementation for this and not make this a part of `given`. What do you think ?

**Comment by dblock** [comment]
I think you should try it and we can see what it looks like!
